Sources/PitchSpeller/Data Structures/EdgeLike.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  EdgeLike.swift
+//  PitchSpeller
+//
+//  Created by Benjamin Wetherfield on 7/1/18.
+//
+
+protocol EdgeLike: Weighted & NodePair {
```

Comment:
Could we get away with just calling this `Edge` if we know that conforming types have to be defined further?

Reply:
I hear ya! My issue was with wanting something like:

`public struct Graph<Node: Hashable, Edge: Edge & Hashable>`, but I think the compiler doesn't particularly enjoy that. On reflection, the `EdgeLike` protocol is basically only there to give us the conditional `.reversed` feature, but that's not giving us huge gains, as I can basically just put that in DirectedEdge without much difference...!

Reply:
The thing with this `EdgeLike` construction is that it kind of enforces that graphs are weighted too so I think I will do away with it